### PR TITLE
Contact endpoints

### LINKED ISSUE #12: Create a Contact
POST 	/api/v1/contacts

### PR DESCRIPTION
**Pull request by amsmyth1**
### What Changed?
- Add Create Contact Endpoint
- Update Contact Details to include contact roles
### What's Next?
- Create Incident Contact
### Known Issues?
- Need to refactor, but in the interest of time I am continuing to Incident Contact for the Front End Team
### Test Coverage?
- 100%
- Edge case testing should be added and more sad path
### README & Documentation
- Create Contact added to API section
- Update json response for IncidentContacts

### Closed Issues
- Close #12

### CODE REVIEW

File: app/models/incident_contact.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class IncidentContact < ApplicationRecord
+  belongs_to :incident
+  belongs_to :contact
+
+  validates_presence_of :incident_id, :contact_id
```
Comment (klatour324) [inline app/models/incident_contact.rb:5]: May need to add in here with the update to the `incidents_contacts` table that it `validates_presence_of :title`

File: spec/models/incident_contact_spec.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe IncidentContact, type: :model do
+  describe "relationships" do
+    it { should belong_to :incident }
+    it { should belong_to :contact }
```
Comment (klatour324) [inline spec/models/incident_contact_spec.rb:6]: And in here `it { should validate_presence_of :title }

**Review by klatour324 — COMMENTED** [review]
Looking good just a couple of minor fixes after you made the migration for `title` on `incidents_contacts` otherwise you are doing tremendous rolling these out 💪.

**Review by klatour324 — APPROVED** [review]
Looks great! Thanks for handling that so quickly!

**Review by jmichlitsch — APPROVED** [review]
